src/games/dama/instance.rs

```diff
@@ -0,0 +1,119 @@
+
+#[derive(Debug)]
+pub(crate) struct Instance {
+    pub(crate) timeout: Duration,
+    pub(crate) pace: Duration,
+    pub(crate) rng: StdRng,
+}
+
+#[async_trait]
+impl game::Instance for Instance {
+    async fn start(&mut self, players: HashMap<String, DuplexStream>, mut spectators: WriteHalf<DuplexStream>,) {
+        let mut giocatore = Player::from(players, &mut self.rng);
+        assert_eq!(giocatore.len(), 2);
+
+        // Invio i nomi dei giocatori
+        for i in 0..2 {
+            lnout2!(giocatore[0].output, &giocatore[i].name);
+            lnout2!(giocatore[1].output, &giocatore[i].name);
+            lnout2!(spectators, &giocatore[i].name);
+        }
+
+        lnout2!(giocatore[0].output, "Avvio la partita di dama...");
+        lnout2!(giocatore[1].output, "Avvio la partita di dama...");
+        lnout2!(spectators, "Avvio la partita di dama...");
+
+        let mut damiera: Vec<Vec<&str>> = vec![vec![" ", "n", " ", "n", " ", "n", " ", "n"],
+                                               vec!["n", " ", "n", " ", "n", " ", "n", " "],
+                                               vec![" ", "n", " ", "n", " ", "n", " ", "n"],
+                                               vec![" ", " ", " ", " ", " ", " ", " ", " "],
+                                               vec![" ", " ", " ", " ", " ", " ", " ", " "],
+                                               vec!["b", " ", "b", " ", "b", " ", "b", " "],
+                                               vec![" ", "b", " ", "b", " ", "b", " ", "b"],
+                                               vec!["b", " ", "b", " ", "b", " ", "b", " "]
+                                              ];
+
+        // Stampo la damiera
+        logic::stampa_damiera(damiera.clone(), &mut giocatore, &mut spectators).await;
+
+        // Setto chi inizzia a giocare
+        let mut turno_binaco: bool = true;
+
+        // Creo il vettore che conterrà il percorso valido
+        let mut percorso_valido: Vec<String> = Vec::new();
+
+        // Inizia sempre la partita il secondo giocatore che si conette
+        _ = giocatore[0].output.write(("Sei i Bianchi\n").as_bytes()).await;
+        _ = giocatore[1].output.write(("Sei i Neri\n").as_bytes()).await;
+        _ = spectators.write(("Il giocatore ".to_owned() + &giocatore[0].name + " è i bianchi e il giocatore " + &giocatore[1].name + " è i neri.\n").as_bytes()).await;
+
+        // Avvio il gioco
+        while !logic::partita_in_corso(damiera.clone(), &mut giocatore, &mut spectators).await {
+            
+            if turno_binaco { // Bianchi
+                _ = giocatore[0].output.write(("Turno bianco!\nE' il tuo turno.\n\n").as_bytes()).await;
+                _ = giocatore[1].output.write(("Turno bianco!\n").as_bytes()).await;
+                _ = spectators.write(("Turno bianco!\n").as_bytes()).await;
+
+                // Controllo chi deve muovere le pedine bianche 
+                if giocatore[0].name.starts_with("ServerBot$")
```

Comment:
Non puoi assumere niente sul bot, per te deve essere come un qualunque altro giocatore. Chi è il bianco e chi è il nero deve essere o determinato in modo random e comunicato ai giocatori, oppure scelto tramite un qualche parametro a priori.